src/python/botan3.py

```diff
@@ -1825,3 +1831,88 @@ def srp6_client_agree(username, password, group, hsh, salt, b, rng):
                                                                     rng.handle_(),
                                                                     a, al,
                                                                     k, kl))
+
+def zfec_encode(k, n, input_bytes):
+    """
+    ZFEC-encode an input message according to the given parameters
+
+    :param int k: the number of shares required to recover the original
+    :param int n: the total number of shares
+    :param bytes input_bytes: the input message, in bytes
+
+    :returns: n arrays of bytes, each one containing a single share
+    """
+    input_size = len(input_bytes)
+    # note: the C++ code checks that input_bytes is a multiple of k
+    outsize = input_size // k
+    p_p_nbytes = c_char_p * n
+
+    # allocate memory for the outputs (create_string_buffer makes bytes)
+    outputs = [
+        create_string_buffer(outsize)
+        for a in range(n)
+    ]
+
+    c_outputs = p_p_nbytes(*[
+        addressof(output)
+        for output in outputs
+    ])
+
+    # actual C call
+    x = _DLL.botan_zfec_encode(
+        c_size_t(k), c_size_t(n), input_bytes, c_size_t(input_size), c_outputs
+    )
+    if x != 0:
+        raise RuntimeError(
+            "Unexpected error, code={}".format(x)
+        )
+    return [output.raw for output in outputs]
+
+
+def zfec_decode(k, n, indexes, inputs):
+    """
+    ZFEC decode
+
+    :param int k: the number of shares required to recover the original
+    :param int n: the total number of shares
+    :param list[int] indexes: which of the shares are we giving the decoder
+    :param list[bytes] inputs: the input shares (e.g. from a previous
+        call to zfec_encode) which all must be the same length
+
+    :returns: a list of bytes containing the original shares decoded
+        from the provided shares (in `inputs`)
+    """
+
+    p_size_t = c_size_t * len(indexes)
+    c_indexes = p_size_t(*[c_size_t(index) for index in indexes])
+    p_p_nbytes = c_char_p * n
+    c_inputs = p_p_nbytes(*[c_char_p(inp) for inp in inputs])
+    # all inputs must be the same length
+    share_size = len(inputs[0])
+    for i in inputs:
+        if len(i) != share_size:
+            raise ValueError(
+                "Share size mismatch: {} != {}".format(len(i), share_size)
+            )
+
+    # allocate memory for our outputs (create_string_buffer creates
+    # bytes)
+    outputs = [
+        create_string_buffer(share_size)
+        for _ in range(k)
+    ]
+    p_p_kbytes = c_char_p * k
+    c_outputs = p_p_kbytes(*[
+        addressof(output)
+        for output in outputs
+    ])
+
+    # actual C call
+    x = _DLL.botan_zfec_decode(
+        c_size_t(k), c_size_t(n), c_indexes, c_inputs, c_size_t(share_size), c_outputs
+    )
+    if x != 0:
```

Comment:
Similarly this check should not be necessary.

Reply:
Fixed in d251adf1b